### PR TITLE
door lock rando support

### PR DESCRIPTION
**Pull request by duncathan**
woo

beam combo blast shields:
image
image
image

### CODE REVIEW

File: open_prime_rando/echoes/dock_lock_rando/dock_type.py [hunk 1]
```diff
@@ -1,16 +1,323 @@
...
+            ACTV -> blast shield, DCTV
+        """
+        super().patch_door(editor, world_name, area_name, dock_name)
+        mrea = self.get_mrea(editor, world_name, area_name)
+        _door = self.get_door_from_dock_index(mrea, self.get_dock_index(world_name, area_name, dock_name))
+        door = _door.get_properties_as(Door)
+
+        default = mrea.get_layer("Default")
+        
+        _sound = default.add_instance("SOND", "Metal Door Lock Breaking")
```
Comment (henriquegemignani) [inline open_prime_rando/echoes/dock_lock_rando/dock_type.py:174]: Wouldn't it be better to use `add_instance_with`? Could be something like this:
```python
sound = default.add_instance_with(Sound(
    editor_properties=EditorProperties(
        name="Metal Door Lock Breaking",
        transform=door.editor_properties.transform,
    ),
    sound=948,
    max_audible_distance=150.0,
    surround_pan=SurroundPan(surround_pan=1.0),
))
```

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

File: open_prime_rando/dynamic_schema.py [hunk 2]
```diff
@@ -9,13 +9,13 @@ def expand_schema(base_schema: dict, editor: PatcherEditor) -> dict:
...
     world_props = schema["properties"]["worlds"]["properties"] = {}
-    for world, mlvl_id in open_prime_rando.echoes.asset_ids.world.NAME_TO_ID.items():
+    for world, mlvl_id in open_prime_rando.echoes.asset_ids.world.NAME_TO_ID_MLVL.items():
         world_def = copy.deepcopy(schema["$defs"]["world"])
         world_props[world] = world_def
 
         mlvl = editor.get_mlvl(mlvl_id)
         area_props = {}
-        world_def["properties"]["areas"] = {"type": "object", "additionalProperties": False, "properties": area_props}
+        world_def["properties"]["areas"] = {"type": "object", "additionalProperties": True, "properties": area_props}
```
[the remark this answers is not in the document]
Comment (duncathan) [reply open_prime_rando/dynamic_schema.py:18]: elevators

File: open_prime_rando/echoes/dock_lock_rando/dock_type.py (hunk 1 above)
Comment (duncathan) [reply open_prime_rando/echoes/dock_lock_rando/dock_type.py:174]: makes sense. I'll need to leave the memory relay as-is tho

File: open_prime_rando/echoes/dock_lock_rando/dock_type.py [hunk 3]
```diff
@@ -1,16 +1,323 @@
...
+        lock.collision_box = self.shield_collision_box
+        lock.collision_offset = self.shield_collision_offset
+        lock.vulnerability = self.vulnerability
+        lock.model = model
+        lock.actor_information.scannable.scannable_info0 = self.get_patched_scan(editor, world_name, area_name)
+        _lock.set_properties(lock)
+
+        _lock.add_connection("DEAD", "PLAY", _sound)
+        _lock.add_connection("DEAD", "PLAY", _streamed)
+        # _lock.add_connection("DEAD", "ACTV", _gibs)
```
Comment (duncathan) [inline open_prime_rando/echoes/dock_lock_rando/dock_type.py:221]: WHOOPS LOL this is why things looked slightly weird in-game

File: open_prime_rando/echoes/dock_lock_rando/dock_type.py [hunk 4]
```diff
@@ -1,16 +1,323 @@
 import dataclasses
+import functools
+import logging
+from open_prime_rando.echoes.dock_lock_rando.map_icons import DoorMapIcon
+from open_prime_rando.patcher_editor import PatcherEditor
+from open_prime_rando.echoes.asset_ids import *
```
[the remark this answers is not in the document]
Comment (henriquegemignani) [reply open_prime_rando/echoes/dock_lock_rando/dock_type.py:6]: I particularly like the namespacing, but i'm fine with linters won't complain in that case

File: open_prime_rando/echoes/dock_lock_rando/dock_type.py (hunk 3 above)
Comment (duncathan) [reply open_prime_rando/echoes/dock_lock_rando/dock_type.py:221]: that's true, but I'd rather it be done much more deliberately than this lol

File: open_prime_rando/echoes/custom_assets/doors/custom_door_lock_template.pdn
[the remark this answers is not in the document]
Comment (duncathan) [reply open_prime_rando/echoes/custom_assets/doors/custom_door_lock_template.pdn:1]: I do like having it right next to the files based on it. is there a good way to exclude it from the wheel but leave it in that folder?

[the remark this answers is not in the document]
Comment (henriquegemignani) [reply open_prime_rando/echoes/custom_assets/doors/custom_door_lock_template.pdn:1]: `MANIFEST.in` I guess?

File: open_prime_rando/dynamic_schema.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (henriquegemignani) [reply open_prime_rando/dynamic_schema.py:18]: oh, add a FIXME

[the remark this answers is not in the document]
Comment (duncathan) [reply open_prime_rando/dynamic_schema.py:18]: had already done that locally by the time you commented lol

File: open_prime_rando/echoes/dock_lock_rando/dock_type.py [hunk 5]
```diff
@@ -1,16 +1,336 @@
...
+
+@dataclasses.dataclass(kw_only=True)
+class VanillaBlastShieldDoorType(BlastShieldDoorType):
+    def remove_blast_shield(self, editor: PatcherEditor, world_name: str, area_name: str, dock_name: str):
+        mrea = self.get_mrea(editor, world_name, area_name)
+        door = self.get_door_from_dock_index(mrea, self.get_dock_index(world_name, area_name, dock_name))
+
+        relay = None
+        for connection in door.connections:
+            if connection.state == "OPEN" and connection.message == "ACTV":
```
Comment (henriquegemignani) [inline open_prime_rando/echoes/dock_lock_rando/dock_type.py:264]: I believe these broke with the new api

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

**Review by henriquegemignani — CHANGES_REQUESTED** [review]
increment rds requirement in setup.cfg

**Review by henriquegemignani — APPROVED** [review]

**Pull request merged**